src/main/kotlin/no/nav/syfo/personstatus/PersonoversiktStatusService.kt

```diff
@@ -116,6 +117,27 @@ class PersonoversiktStatusService(
             isAktivVurdering = isAktivVurdering,
         )
 
+    suspend fun updateNavnOrFodselsdatoWhereMissing(updateLimit: Int) {
+        val personStatuser = personoversiktStatusRepository.getPersonstatusesWithoutNavnOrFodselsdato()
+        val personidenter = personStatuser.map { PersonIdent(it.fnr) }
+
+        val pdlPersons = pdlClient.getPersons(personidenter = personidenter)?.hentPersonBolk ?: emptyList()
+        val pdlPersonsById = pdlPersons.associateBy { it.ident }
+        val editedPersonStatuser = personStatuser.mapNotNull { personStatus ->
+            val pdlPerson = pdlPersonsById[personStatus.fnr]
+            val fullName = pdlPerson?.person?.fullName()
+            val fodselsdato = pdlPerson?.person?.foedselsdato?.first()?.foedselsdato
+
+            val isUpdate = fullName != null || fodselsdato != null
+            if (isUpdate) {
+                personStatus.updatePersonDetails(navn = fullName, fodselsdato = fodselsdato)
+            } else {
+                null
+            }
+        }
+        personoversiktStatusRepository.updatePersonstatusesWithNavnAndFodselsdato(editedPersonStatuser.take(updateLimit))
```

Comment:
Tror det vil være fordelaktig for ytelsen å bruke `limit` i sql-en i repository i stedet for å mappe hele resultatet av spørringen til objekter og så begrense.

Reply:
Ja, det er bedre å bruke LIMIT i sql'en